test/lightning/jobs_test.exs

```diff
@@ -284,6 +284,26 @@ defmodule Lightning.JobsTest do
       assert {:ok, %Job{}} = Jobs.delete_job(job)
       assert_raise Ecto.NoResultsError, fn -> Jobs.get_job!(job.id) end
     end
+
+    test "delete_job/1 can't delete job with downstream jobs" do
+      job = job_fixture()
+
+      {:ok, %Job{} = _} =
+        Jobs.create_job(%{
+          body: "some body",
+          enabled: true,
+          name: "some name",
+          adaptor: "@openfn/language-common",
+          trigger: %{type: "on_job_success", upstream_job_id: job.id},
+          workflow_id: job.workflow_id
+        })
+
+      {:error, changeset} = Jobs.delete_job(job)
+
+      assert {:trigger_id,
+              {"This job is associated with downstream jobs",
+               [constraint: :foreign, constraint_name: "jobs_trigger_id_fkey"]}} in changeset.errors
```

Comment:
By the way there is a helper for these kinds of matches.

Example from `AccountsTest`:

```
assert %{
  password: ["can't be blank"],
  email: ["can't be blank"]
} = errors_on(changeset)
```